src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/ComponentBasedApplicationBase.cs

```diff
@@ -0,0 +1,208 @@
+﻿/*
+ * Copyright (c) 2019 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+
+namespace Tizen.Applications.ComponentBased.Common
+{
+    /// <summary>
+    /// The class for supporting multi-components based application model.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public class ComponentBasedApplicationBase : Application
```

Comment:
Why do you need this middle layer between `Application` and `ComponentBasedApplication`?
This class only has one subclass(ComponentBasedApplication)

Reply:
ComponentBasedApplicationBase has to manage a totally different lifecycle with Application which is needed by FrameComponent, ServiceComponent etc..

And ComponentBasedApplication is EFL base application. Dali base application can be added in the future.

Reply:
As your comment. 
ComponentBasedApplicationBase -> ComponentBasedApplication 
ComponentBasedApplication  -> EFLComponentBasedApplication

Reply:
Done